src/Tests/Tests/Test_Vibrato.cpp

```diff
@@ -0,0 +1,296 @@
+#include "MUSI6106Config.h"
+
+#ifdef WITH_TESTS
+#include <cassert>
+#include <cstdio>
+#include <algorithm>
+
+#include "gtest/gtest.h"
+
+#include "Synthesis.h"
+#include "Vector.h"
+#include "ErrorDef.h"
+
+#include "Vibrato.h"
+#include "RingBuffer.h"
+
+namespace vibrato_test {
+    void CHECK_ARRAY_CLOSE(float* buffer1, float* buffer2, int iLength, float fTolerance)
+    {
+        for (int i = 0; i < iLength; i++)
+        {
+            EXPECT_NEAR(buffer1[i], buffer2[i], fTolerance);
+        }
+    }
+    
+class CTestVibrato : public testing::Test
+{
+public:
+    CVibrato            *pVibrato = 0;
+    float**             ppfAudioData = 0;
+    float**             ppfOutputData = 0;
+        
+    float**             ppfTempIn = 0;
+    float**             ppfTempOut = 0;
+        
+    float               fMaxWidthSecs = 0.1;
+    float               fSampleRate = 44100;
+    int                 iNumChannels = 2;
+    int                 iBlockSize = 1024;
+    float               fModFreq = 10;
+    float               fModWidth = 0.05;
+    float               fDelay = 0.05;
+    
+    void SetUp() override
+    {
+        CVibrato::create(pVibrato);
+
+        pVibrato -> CVibrato::init(fDelay, fSampleRate, fModWidth, fModFreq, iNumChannels);
+        
+        ppfAudioData  = new float*[iNumChannels];
+        ppfOutputData = new float*[iNumChannels];
+        
+        ppfTempIn  = new float*[iNumChannels];
+        ppfTempOut = new float*[iNumChannels];
+        
+        float fTestSigFreq = 440.F;
+        
+        for (int i = 0; i < iNumChannels; i++)
+        {
+            ppfAudioData[i] = new float[iBlockSize];
+            ppfOutputData[i] = new float[iBlockSize];
+            
+            ppfTempIn[i]  = new float[iBlockSize];
+            ppfTempOut[i] = new float[iBlockSize];
+
+            CSynthesis::generateSine(ppfAudioData[i], fTestSigFreq, fSampleRate, iBlockSize);
```

Comment:
Nice use of `SetUp()`, `TearDown()`, and library functions to avoid duplication!

Reply:
Thanks!